src/app/main/listing/listing.component.ts

```diff
@@ -32,6 +32,11 @@ export class ListingComponent extends NgxSubFormRemapComponent<OneListing, OneLi
 
   public ListingType = ListingType;
 
+  // it is required to define the formControlName as this is supposed to be a
+  // sub form but in that case we use NgxSubFormRemapComponent for convenience
+  // at the top level and easily create the form + get utilities
+  public formControlName = 'ListingComponent';
```

Comment:
That kind of sucks, but we use the `formControlName` with the validate function and so we need a check to make sure it exists. In the case when we use the lib on the top form component for convenience, the `formControlName` is not present.

Reply:
Refactored.